math/complex_numbers.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2020 @author tjgurwara99
+ * @file
+ *
+ * A basic implementation of Complex Number field as a class with operators
+ * overloaded to accommodate (mathematical) field operations.
+ */
+
+#include <cmath>
+#include <iostream>
+#include <stdexcept>
+#include <cassert>
+
+/**
+ * Class Complex to represent complex numbers as a field.
+ */
+class Complex {
+    // The real value of the complex number
+    double re;
+    // The imaginary value of the complex number
+    double im;
+
+  public:
+    /**
+     * Complex Constructor which initialises the complex number which takes two
+     * arguments.
+     * @param x The real value of the complex number.
+     * @param y The imaginary value of the complex number.
+     */
+    Complex(double x, double y) {
+        this->re = x;
+        this->im = y;
+    }
+
+    /**
+     * Complex Constructor which initialises the complex number with no
+     * arguments.
+     */
+    Complex() { Complex(0.0, 0.0); }
+
+    /**
+     * Member function (getter) to access the class' re value.
+     */
+    double real() const { return this->re; }
+
+    /**
+     * Member function (getter) to access the class' im value.
+     */
+    double imag() const { return this->im; }
+
+    /**
+     * Member function to which gives the absolute value (modulus) of our
+     * complex number
+     * @return \f$ \sqrt{z \dot \bar{z}} \f$ where \f$ z \f$ is our complex
+     * number.
+     */
+    double abs() const {
```

Comment:
please include `double arg() const {` for the argument value of a complex number

Reply:
Oh yeah, I forgot about that😅